<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="initial_x" default="0" />
    <arg name="initial_y" default="0" />
    <arg name="name" default="robot_1" />
    <arg name="id" default="1" />
    <arg name="node_start_delay" default="6.0" />
    <param name="initial_x" value="$(arg initial_x)" />
    <param name="initial_y" value="$(arg initial_y)" />
    <param name="name" value="$(arg name)"/>
    <param name="id" value="$(arg id)"/>
    <remap from="/$(arg name)/$(arg name)/cmd_vel" to="/$(arg name)/cmd_vel"/>
    <remap from="/$(arg name)/$(arg name)/scan" to="/$(arg name)/scan"/>
    <remap from="/$(arg name)/$(arg name)/odom_diffdrive" to="/$(arg name)/odom_diffdrive"/>
    <remap from="/$(arg name)/$(arg name)/go_to_point_switch" to="/$(arg name)/go_to_point_switch" />
    <remap from="/$(arg name)/$(arg name)/wall_follower_switch" to="/$(arg name)/wall_follower_switch" />
    <remap from="/$(arg name)/$(arg name)/camera_pose" to="/$(arg name)/camera_pose" />
    <node pkg="epuck_gazebo" 
        type="follow_wall.py" 
        name="wall_follower" />
    <node 
        pkg="epuck_gazebo" 
        type="go_to_point.py" 
        name="go_to_point" />
    <!--<node 
        pkg="epuck_gazebo" 
        type="camera_pose.py" 
        name="camera_pose" />-->
    <node pkg="epuck_gazebo" 
        type="go_to_camera_2.py" 
        name="go_to_camera" 
        output="screen" 
        launch-prefix="bash -c 'sleep $(arg id); $0 $@' "/>
</launch>